content/hardware/08.edu/solution-and-kits/alvik/tutorials/setting-alvik-arduino-ide/setting-alvik-arduino-ide.md

```diff
@@ -0,0 +1,184 @@
+
+```c++
+Arduino_Alvik alvik;
+```
+
+When using the Alvik livrary we declare an object of the `Arduino_Alvik` class named `alvik` in this case. All interactions with the robot will go through this `alvik` object. As with our `alvik.drive()` command.
+
+```c++
+void setup() {
+  alvik.begin();
+  Serial.begin(115200);
+}
+```
+
+In the `setup()` function, the `alvik.begin()` method initializes the Alvik robot. This will be a necessity on all sketches for the Alvik.
+
+```c++
+void loop() {
+
+  uint8_t u,m,l;
+
+  alvik.get_version(u,m,l); // Gets the firmware version
+  Serial.printf("%d.%d.%d\n", u, m, l);
+  alvik.drive(10, 45);
+  delay(1000); // Waits a second
+  alvik.drive(10, -45);
+  delay(1000); // Waits a second
+
+}
+```
+
+This code continuously retrieves and prints the firmware version thanks to the ```get_version()``` every second while making the Alvik robot's wheels rotate back and forth using the ```drive()``` command.
+
+
+**Complete code:**
+
+```c++
+#include "Arduino_Alvik.h"
+
+Arduino_Alvik alvik;
+
+void setup() {
+  alvik.begin();
+  Serial.begin(115200);
+}
+
+void loop() {
+
+  uint8_t u,m,l;
+
+  alvik.get_version(u,m,l); // Gets the firmware version
+  Serial.printf("%d.%d.%d\n", u, m, l);
+  alvik.drive(10, 45);
+  delay(1000); // Waits a second
+  alvik.drive(10, -45);
+  delay(1000); // Waits a second
+
+}
+```
+
+### Obstacle Avoider Example In C++
+
+A more complex example can be found in the [getting started](https://docs.arduino.cc/tutorials/alvik/getting-started/) guide for the Alvik.
+Due to the functions having same structure and names on both MicroPython and C++ we can easily port the ```obstacle-avoider``` to C++.
+
+Keeping in mind the initialization of the Alvik from the previous example, we can build the example on the Arduino IDE:
+
+```c++
+#include "Arduino_Alvik.h"
+#include <Arduino.h>
+
+Arduino_Alvik alvik;
+
+void setup() {
+  alvik.begin();
+  delay(5000);  // Waiting for the robot to setup
+}
+
+void loop() {
+  float distance = 12.0;
+  float degrees = 45.0;
+  float speed = 10.0;
+
+  float distance_l, distance_cl, distance_c, distance_r, distance_cr;
+  
+  alvik.get_distance(distance_l, distance_cl, distance_c, distance_r, distance_cr);
+  delay(50);
+  
+  Serial.println(distance_c);
+
+  if (distance_c < distance || distance_cl < distance || distance_cr < distance || distance_l < distance || distance_r < distance) {
+    alvik.rotate(degrees);
+  } else {
+    alvik.drive(speed, 0.0);
+  }
+}
+
+```
+
+You can now explore the other included examples that cover more of the Alvik's components and more functions listed on our [API reference](https://docs.arduino.cc/tutorials/alvik/api-overview/).
+
+
+## More Resources (C++)
```

Comment:
```suggestion
## More Resources
```

The resources aren't C++-specific, so there is no point in putting this in the heading.